### DISCUSSION TITLE
Not able to launch demo app

### DISCUSSION THREAD

**Discussion by Sandrinotje213** [AdaptiveMotorControlLab/AmadeusGPT#30] [Q&A] [answered]
Hello, I wanted to try out AmadeusGPT but I do not have much experience with projects like this. Perhaps someone can help me to launch the app, as I feel I am missing something that is obvious.

I have followed the installation page, using Anaconda Prompt (terminal). I created an environment (although I had to remove the apostrophes in the pip install 'amadeusgpt[streamlit]' command), gitcloned the demo files, and went to the correct directory.  No errors appear, until I try to actually launch the app. The terminal does not recognise ' make app'  as a command, nor did ' python -m launch_amadeusGPT' work. The Anaconda terminal then returns the following:

__________________________________________________________________________________________________________________
_Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\ProgramData\Anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\DevNeurosci\Documents\AmadeusGPT\launch_amadeusGPT.py", line 3, in <module>
    from amadeusgpt import app
  File "C:\Users\DevNeurosci\Documents\AmadeusGPT\amadeusgpt\__init__.py", line 8, in <module>
    from amadeusgpt.implementation import AnimalBehaviorAnalysis
  File "C:\Users\DevNeurosci\Documents\AmadeusGPT\amadeusgpt\implementation.py", line 8, in <module>
    from amadeusgpt import utils
  File "C:\Users\DevNeurosci\Documents\AmadeusGPT\amadeusgpt\utils.py", line 12, in <module>
    from amadeusgpt.logger import AmadeusLogger
  File "C:\Users\DevNeurosci\Documents\AmadeusGPT\amadeusgpt\logger.py", line 1, in <module>
    from google.cloud import firestore
ModuleNotFoundError: No module named 'google'_
__________________________________________________________________________________________________________________

I also tried to launch the app in Google Colab, following the 'examples\[simple_amadeusGPT_demo.ipynb' file. I gitcloned the Amadeus GitHub files here as well, altered the file locations to fit the Google Colab. Everything works until I want to run "answer = AMADEUS.chat_iteration(query)". Colab then returns the following:

__________________________________________________________________________________________________________________
_2024-04-19 08:17:12.938 
  Warning: to view this Streamlit app on a browser, run it with the following
  command:

    streamlit run /usr/local/lib/python3.10/dist-packages/colab_kernel_launcher.py [ARGUMENTS]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-12-b7f8e9b38daa>](https://localhost:8080/#) in <cell line: 1>()
----> 1 answer = AMADEUS.chat_iteration(query)
      2 render(answer)

1 frames
[/usr/local/lib/python3.10/dist-packages/amadeusgpt/main.py](https://localhost:8080/#) in load_module_smartly(cls, user_input)
    126     def load_module_smartly(cls, user_input):
    127         sorted_query_results = match_module(user_input)
--> 128         if len(sorted_query_results) == 0:
    129             return None
    130         # query result sorted by most relevant module text

TypeError: object of type 'NoneType' has no len()_
__________________________________________________________________________________________________________________

I then attempted to run !streamlit run /usr/local/lib/python3.10/dist-packages/colab_kernel_launcher.py [query] after defining 'query' , but this seems to just create a loop (waited for 5 minutes) after saying the connection was succesful (IPs are masked):
__________________________________________________________________________________________________________________
_Collecting usage statistics. To deactivate, set browser.gatherUsageStats to false.

  You can now view your Streamlit app in your browser.

  Network URL: http://[ip1]/
  External URL: http://[ip2]/_

__________________________________________________________________________________________________________________

Any help would be appreciated.

**Comment by yeshaokai** [comment, marked as answer]
@Sandrinotje213 

Thanks for reporting the issue! We are testing a new release that should fix this and it's much better than the previous one. Feel free to test it and give us feedback!

https://github.com/AdaptiveMotorControlLab/AmadeusGPT/tree/release-0.1.0

**Comment by laurelrr** [comment]
Hello, 
I thought I would chime in here but let me know if you prefer me to open an issue.  I have just installed amadeusgpt through conda and exported my open ai key as instructed in the conda environment.  I am trying to run the jupyter notebook EPM_demo and I get a message `not able to import cebra` right after I run the first cell.  Is this a dependency that I should install separately?    

But the main error comes after the "Set the scene number..." cell:
```
Project created at results. Results will be saved to results
The project will load video files (*.mp4) and optionally keypoint files from ../examples/EPM
A copy of the project config file is saved at results/config.yaml
{'data_info': {'data_folder': '../examples/EPM',
               'result_folder': 'results',
               'video_suffix': '.mp4'},
 'keypoint_info': {'body_orientation_keypoints': {'animal_center': 'neck',
                                                  'neck': 'nose',
                                                  'tail_base': 'tail_base'},
                   'head_orientation_keypoints': {'neck': 'neck',
                                                  'nose': 'nose'},
                   'include_confidence': False,
                   'use_3d': False},
 'llm_info': {'gpt_model': 'gpt-4o',
              'keep_last_n_messages': 2,
              'max_tokens': 4096,
              'temperature': 0.0},
 'object_info': {'load_objects_from_disk': False, 'use_grid_objects': False},
 'video_info': {'scene_frame_number': 400}}
INFO:httpx:HTTP Request: POST https://api.openai.com/v1/chat/completions "HTTP/1.1 404 Not Found"
INFO:httpx:HTTP Request: POST https://api.openai.com/v1/chat/completions "HTTP/1.1 404 Not Found"
error Traceback (most recent call last):
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/amadeusgpt/analysis_objects/llm.py", line 101, in connect_gpt_oai_1
    response = client.chat.completions.create(**json_data)
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_utils/_utils.py", line 275, in wrapper
    return func(*args, **kwargs)
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/resources/chat/completions.py", line 829, in create
    return self._post(
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_base_client.py", line 1280, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_base_client.py", line 957, in request
    return self._request(
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_base_client.py", line 1061, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.NotFoundError: Error code: 404 - {'error': {'message': 'The model `gpt-4o` does not exist or you do not have access to it.', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}

error Traceback (most recent call last):
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/amadeusgpt/analysis_objects/llm.py", line 101, in connect_gpt_oai_1
    response = client.chat.completions.create(**json_data)
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_utils/_utils.py", line 275, in wrapper
    return func(*args, **kwargs)
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/resources/chat/completions.py", line 829, in create
    return self._post(
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_base_client.py", line 1280, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_base_client.py", line 957, in request
    return self._request(
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_base_client.py", line 1061, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.NotFoundError: Error code: 404 - {'error': {'message': 'The model `gpt-4o` does not exist or you do not have access to it.', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}

INFO:httpx:HTTP Request: POST https://api.openai.com/v1/chat/completions "HTTP/1.1 404 Not Found"
error Traceback (most recent call last):
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/amadeusgpt/analysis_objects/llm.py", line 101, in connect_gpt_oai_1
    response = client.chat.completions.create(**json_data)
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_utils/_utils.py", line 275, in wrapper
    return func(*args, **kwargs)
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/resources/chat/completions.py", line 829, in create
    return self._post(
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_base_client.py", line 1280, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_base_client.py", line 957, in request
    return self._request(
  File "/home/lkeyes/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/openai/_base_client.py", line 1061, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.NotFoundError: Error code: 404 - {'error': {'message': 'The model `gpt-4o` does not exist or you do not have access to it.', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[3], line 17
      4 kwargs = {
      5     "keypoint_info.body_orientation_keypoints.neck" : "nose",
      6     "keypoint_info.body_orientation_keypoints.tail_base" : "tail_base",
   (...)
     10     "video_info.scene_frame_number" : scene_frame_number,
     11 }
     13 config = create_project(data_folder = "../examples/EPM",
     14                         result_folder = "results",
     15                         **kwargs
     16                         )
---> 17 amadeus = AMADEUS(config, use_vlm=True)
     18 video_file_paths = amadeus.get_video_file_paths()
     19 print (video_file_paths)

File ~/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/amadeusgpt/main.py:64, in AMADEUS.__init__(self, config, use_vlm, movie2keypoint_func)
     62 # can only do this after the register process
     63 if use_vlm:
---> 64     self.sandbox.configure_using_vlm()

File ~/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/amadeusgpt/programs/sandbox.py:227, in Sandbox.configure_using_vlm(self)
    224 for identifier, analysis in self.analysis_dict.items():
    226     scene_image = analysis.visual_manager.get_scene_image()
--> 227     json_obj = self.llms["visual_llm"].speak(self, scene_image)
    228     self.meta_info[identifier] = json_obj
    229     # configure meta info on the analysis managers

File ~/anaconda3/envs/amadeusgpt/lib/python3.10/site-packages/amadeusgpt/analysis_objects/llm.py:263, in VisualLLM.speak(self, sandbox, image)
    256 self.update_history(
    257     "user",
    258     "here is the image",
    259     multi_image_content=multi_image_content,
    260     in_place=True,
    261 )
    262 response = self.connect_gpt(self.context_window, max_tokens=2000)
--> 263 text = response.choices[0].message.content.strip()
    265 print("description of the image frame provided")
    266 print(text)

AttributeError: 'NoneType' object has no attribute 'choices'
```

**Reply by laurelrr** [reply]
note: version is 0.1.3

**Reply by laurelrr** [reply]
also, minor typo in your link to the notebooks `We provide example notebooks [here](https://github.com/AdaptiveMotorControlLab/AmadeusGPT/blob/main/notebook)!`
this seems to have changed

**Reply by yeshaokai** [reply]
@laurelrr Thanks for trying our code. From the error message, it suggests that you haven't put an openai API key in the beginning of the notebook, which is required for using openAI's GPT for the vision language understanding.
Here is a guide for creating the API key: https://platform.openai.com/docs/quickstart

**Reply by laurelrr** [reply]
Do I also need to have a paid subscription at openAI to use this program?

**Reply by yeshaokai** [reply]
@laurelrr Not really. Subscription and API are two different things. API is "pay-as-you-go". You will most likely be charged for fraction of cents when using this program. I also print the estimated cost in the outputs.

**Reply by laurelrr** [reply]
OK, I think the error was due to a payment issue. After saving my credit card and purchasing an initial $10 of credits, it seems to be working.  I think I was initially confused because this API service is not included with the chatGPT subscription I have.  You have to purchase this separately.
